File: app/src/main/java/com/gs/buluo/store/presenter/BankCardPresenter.java
package com.gs.buluo.store.presenter;

import com.gs.buluo.store.R;
import com.gs.buluo.store.bean.ResponseBody.CardResponse;
import com.gs.buluo.store.model.MoneyModel;
import com.gs.buluo.store.view.impl.ICardView;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

/**
 * Created by hjn on 2016/11/23.
 */
public class BankCardPresenter extends BasePresenter<ICardView> {
    MoneyModel moneyModel;

    public BankCardPresenter(){
        moneyModel=new MoneyModel();
    }

    public void getCardList(String uid){
        moneyModel.getCardList(uid, new Callback<CardResponse>() {
            @Override
            public void onResponse(Call<CardResponse> call, Response<CardResponse> response) {
                if (response.body()!=null&&response.body().code==200){
                    mView.getCardInfoSuccess(response.body().data);
                }else {
                    mView.showError(R.string.connect_fail);
                }
            }

            @Override
            public void onFailure(Call<CardResponse> call, Throwable t) {
                mView.showError(R.string.connect_fail);
            }
        });
    }
}

File: app/src/main/java/com/gs/buluo/store/presenter/BankCardPresenter.kt
package com.gs.buluo.store.presenter

import com.gs.buluo.common.network.BaseResponse
import com.gs.buluo.common.network.BaseSubscriber
import com.gs.buluo.store.bean.BankCard
import com.gs.buluo.store.kotlin.presenter.KotBasePresenter
import com.gs.buluo.store.network.MoneyApis
import com.gs.buluo.store.network.TribeRetrofit
import com.gs.buluo.store.view.impl.ICardView
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

/**
 * Created by hjn on 2016/11/23.
 */
class BankCardPresenter(var mView: ICardView) : KotBasePresenter() {
    fun getCardList(uid: String) {
        TribeRetrofit.getInstance().createApi(MoneyApis::class.java).getCardList(uid)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object : BaseSubscriber<BaseResponse<List<BankCard>>>() {
                    override fun onNext(response: BaseResponse<List<BankCard>>) {
                        mView.getCardInfoSuccess(response.data)
                    }
                })
    }
}

